app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorDetailView.java

```diff
@@ -287,4 +290,21 @@ private void setupTextView(View rootView, int id, String text) {
     textView.setText(text);
     textView.setKeyListener(null);
   }
+
+  private void setupImageView(View rootView, Favor favor) {
+    String url = favor.getPictureUrl();
+    if (url != null) {
```

Comment:
In the observer we might need to add a check that the favorId==favor.getId()  above when a new favor is observed

Reply:
Actually it's fine. Shouldn't be a problem